scripts/Diffraction/single_crystal/base_sx.py

```diff
@@ -614,6 +618,155 @@ def remove_non_integrated_peaks(peaks):
+                        rf"d={np.round(pk.getDSpacing(), 2)} $\AA$"
+                    )
+                    fig.tight_layout()
+                    pdf.savefig(fig)
+                    close(fig)
+
+                if ws_cut is not None:
+                    mantid.DeleteWorkspace(ws_cut)
+        except OSError:
+            raise RuntimeError(
+                f"OutputFile ({filename}) could not be opened - please check it is not open by "
+                f"another programme and that the user has permission to write to that directory."
+            )
+
+    @staticmethod
+    def _bin_MD_around_peak(wsMD, pk, peak_shape, nbins_max, extent, frame_to_peak_centre_attr):
+        shape_info = json_loads(peak_shape.toJSON())
+        if peak_shape.shapeName().lower() == "spherical":
+            BaseSX.convert_spherical_representation_to_ellipsoid(shape_info)
+        # get radii
+        radii = np.array([shape_info[f"radius{iax}"] for iax in range(3)])
+        bg_inner_radii = np.array([shape_info[f"background_inner_radius{iax}"] for iax in range(3)])
+        bg_outer_radii = np.array([shape_info[f"background_outer_radius{iax}"] for iax in range(3)])
+        isort = np.argsort(radii)
+        imax = isort[-1]
+        # eignevectors of ellipsoid
+        evecs = np.zeros((3, 3))
+        for iax in range(len(radii)):
+            evec = np.array([float(elem) for elem in shape_info[f"direction{iax}"].split()])
+            evecs[:, iax] = evec / np.linalg.norm(evec)
+        # center and translation
+        translation = np.array([shape_info[f"translation{iax}"] for iax in range(3)])
+        cen = getattr(pk, frame_to_peak_centre_attr)()
+        cen = np.matmul(evecs.T, np.array(cen) + translation)
+        # get extents
+        box_lengths = bg_outer_radii if np.all(bg_outer_radii > 1e-10) else radii
+        box_lengths = box_lengths * extent
+        extents = np.vstack((cen - box_lengths, cen + box_lengths))
+        # get nbins along each axis
+        nbins = np.array([int(nbins_max * radii[iax] / radii[imax]) for iax in range(len(radii))])
+        # ensure minimum of 3 bins inside radius along each dim
+        min_nbins_in_radius = np.min(nbins * radii / box_lengths)
+        if min_nbins_in_radius < 3:
+            nbins = nbins * 3 / min_nbins_in_radius
+        # call BinMD
+        ws_cut = mantid.BinMD(
+            InputWorkspace=wsMD,
+            OutputWorkspace="__ws_cut",
+            AxisAligned=False,
+            BasisVector0=r"Q$_0$,unit," + ",".join(np.array2string(evecs[:, 0], precision=6).strip("[]").split()),
+            BasisVector1=r"Q$_1$,unit," + ",".join(np.array2string(evecs[:, 1], precision=6).strip("[]").split()),
+            BasisVector2=r"Q$_2$,unit," + ",".join(np.array2string(evecs[:, 2], precision=6).strip("[]").split()),
+            OutputExtents=extents.flatten(order="F"),
+            OutputBins=nbins.astype(int),
+            EnableLogging=False,
+        )
```

Comment:
I always wondered about this, I tend to store in the ADS and use `DeleteWorkspace` so I'm sure the memory is cleared? Do you know when the memory would be cleared if it wasn't in the ADS?

Reply:
If it isn't in the ADS, it should be cleared when the number of smart pointer reference counts gets to zero. I think this is the same as it would be if it was in the ADS. When its in the ADS there is probably 1 extra on the count. All `DeleteWorkspace` does is remove 1 from the count by removing the workspace smart pointer from the ADS. In the case of `StoreInADS=False`, we avoid having this extra 1 on the count all together